store/db.go

```diff
@@ -49,15 +51,17 @@ func newDBLock(dirPath string) (*dbLock, error) {
 }
 
 func (dl *dbLock) lock() error {
+	dl.Lock()
 	if err := dl.fl.ExclusiveLock(); err != nil {
+		dl.Unlock()
 		return err
 	}
-	dl.Lock()
 	return nil
 }
 
 func (dl *dbLock) unlock() error {
 	if err := dl.fl.Unlock(); err != nil {
+		dl.Unlock()
 		return err
 	}
 	dl.Unlock()
```

Comment:
You want to always unlock the mutex guarding the fileLock second since it's the "outer lock". Unlocking it first would leave the file lock unprotected.

I do think this function would be slightly clearer with a named error, e.g.

```
// Named error, err
err = dl.fl.Unlock()
dl.Unlock
```